### PR TITLE
Redux thunks Introduction

### PR DESCRIPTION
**Pull request by kulkarniankita**
The content slides for thunks now include:

- What is a Thunk?
- Why do we need to use a middleware?
- Redux Thunks
- Connect Redux and Thunks
- Dispatch actions using Thunks

💃 😄

### CODE REVIEW

File: src/slides/redux-thunks/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  ReplFrameSlide,
+} from '../../tools';
+
+/* eslint import/no-webpack-loader-syntax: off */
+import * as reduxThunksConnectExample from '!raw-loader!./thunks-redux-connect-example';
+import * as reduxThunksBasicExample from '!raw-loader!./thunks-basic-example';
+
+const Slide1 = BasicSlideMaker(
+  `What is a Thunk?`,
+  `Thunk is basically a function`,
```
Comment (pkanal) [inline src/slides/redux-thunks/index.js]: It's a special name for a function that is returned by another function.

File: src/slides/redux-thunks/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  `It is a functional programming technique used to delay a task`,
+  `Perform work later rather than doing it now and instead, return a function or an expression`,
+);
+
+const Slide2 = BasicSlideMaker(
+  `Why do we need to use a middleware?`,
+  `Actions are plain objects and we use action creators so we can pass in specific information to that action, for example`,
+  `const ADD_TODO = (text) => { type: "ADD_TODO", text };`,
+  `What if we want to make our actions do something fun such as do some "asyc" but we can't perform async network calls in reducers as they need to be pure functions and we don't want side-effects.`,
```
Comment (pkanal) [inline src/slides/redux-thunks/index.js]: small typo: "asyc" --> "async"

File: src/slides/redux-thunks/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  `Hence we use a middleware such as Redux thunks.`,
```
Comment (pkanal) [inline src/slides/redux-thunks/index.js]: `redux-thunk`

File: src/slides/redux-thunks/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+);
+
+const Slide5 = BasicCodeSlideMaker(
+  reduxThunksBasicExample,
+  16,
+  'Dispatch actions using Thunks',
+);
+
+const Slide6 = ReplFrameSlide(
+  'https://repl.it/@ShaunLloyd1/redux-thunks-in-react',
```
Comment (pkanal) [inline src/slides/redux-thunks/index.js]: Could you replace the link with `https://repl.it/@AbdellaPurvi/redux-thunks-in-react?lite=true`? Sorry I meant to fork this over earlier!

File: src/slides/redux-thunks/index.js (hunk 2 above)
Comment (kulkarniankita) [reply src/slides/redux-thunks/index.js]: lol whoops 😄

**Comment by pkanal** [comment]
Woooo shipped ✨ 🚀

**Pull request merged**